google/generativeai/types/model_types.py

```diff
@@ -183,16 +186,71 @@ class TuningExampleDict(TypedDict):
     output: str
 
 
-TuningExampleOptions = Union[TuningExampleDict, glm.TuningExample, tuple[str, str]]
+TuningExampleOptions = Union[TuningExampleDict, glm.TuningExample, tuple[str, str], list[str]]
 TuningDataOptions = Union[
-    glm.Dataset, Iterable[TuningExampleOptions]
-]  # TODO(markdaoust): csv, json, pandas, np
+    pathlib.Path, str, glm.Dataset, Mapping[str, Iterable[str]], Iterable[TuningExampleOptions]
+]  # TODO(markdaoust): json-URLS, gs:// URLS .. use etils.epath?
 
 
-def encode_tuning_data(data: TuningDataOptions) -> glm.Dataset:
+def encode_tuning_data(
+    data: TuningDataOptions, input_key="text_input", output_key="output"
+) -> glm.Dataset:
+    if isinstance(data, str):
+        # Strings are either URLs or system paths.
+        if re.match("^\w+://\S+$", data):
+            data = _normalize_url(data)
+        else:
+            # Normalize system paths to use pathlib
+            data = pathlib.Path(data)
+
+    if isinstance(data, pathlib.Path):
+        if data.suffix.lower() == ".json":
+            with open(data) as f:
+                data = json.load(f)
+
+    if isinstance(data, (str, pathlib.Path)):
+        import pandas as pd
+
+        data = pd.read_csv(data)
```

Comment:
MakerSuite supports CSVs and Sheets without header rows - can we do that here too?

Not a big deal I guess - the error handling looks pretty thorough, so it should be clear to users what the issue is. Maybe just add a note to the `create_tuned_model` docstring saying a header is required?

Reply:
> MakerSuite supports CSVs and Sheets without header rows - can we do that here too?

How do we know which column to use?

csv.DictReader requires a header.